cf-agent/verify_users_pam.c

```diff
@@ -622,6 +630,45 @@ static bool ChangePasswordHashUsingLckpwdf(const char *puser, const char *passwo
 }
 #endif // defined(HAVE_LCKPWDF) && defined(HAVE_ULCKPWDF)
 
+static bool ExecuteUserCommand(const char *puser, const char *cmd, size_t sizeof_cmd,
+                               const char *action_msg, const char *cap_action_msg)
+{
+    if (strlen(cmd) >= sizeof_cmd - 1)
+    {
+        // Instead of checking every StringAppend call, assume that a maxed out
+        // string length overflowed the string.
+        Log(LOG_LEVEL_ERR, "Command line too long while %s user '%s'", action_msg, puser);
+        return false;
+    }
+
+    Log(LOG_LEVEL_VERBOSE, "%s user '%s'. (command: '%s')", cap_action_msg, puser, cmd);
+
+    const int status = system(cmd);
+    if (!WIFEXITED(status) || WEXITSTATUS(status) != 0)
```

Comment:
Apparently, on RHEL 5 this cannot be `const`, can you try to remove the `const` keyword here?

Reply:
Output:

```
11:34:32 verify_users_pam.c: In function 'ExecuteUserCommand':
11:34:32 verify_users_pam.c:647: error: assignment of read-only member '__in'
11:34:32 verify_users_pam.c:647: error: assignment of read-only member '__in'
11:34:32 gmake[2]: *** [verify_users_pam.lo] Error 1
11:34:32 gmake[2]: Target `all' not remade because of errors.
11:34:32 Making all in cf-execd
```

Reply:
@xaxo05 Did you click resolved on accident? We need the code to compile on RHEL 5 before merging it, so this needs to be fixed.